### ISSUE TITLE
Add customizable optimizer for the fit method

### ISSUE THREAD

**Issue by D-Exposito** [CESGA-Quantum-Spain/cunqa#242]
Add Adam optimizer to allow dynamic learning rate: pennylane's adam optimizer would make the training better, by avoiding ticktacking too much around the minimun due to large gradients.

Other optimizers would also provide richness of options to explore.
